packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -710,59 +712,17 @@ export class GarbageCollector implements IGarbageCollector {
          * GC state and updates their inactive or sweep ready state.
          */
         this.initializeGCStateFromBaseSnapshotP = new LazyPromise<void>(async () => {
-            const currentReferenceTimestampMs = this.runtime.getCurrentReferenceTimestampMs();
-            /**
-             * If there is no current reference timestamp, skip initialization. We need the current timestamp to track
-             * how long objects have been unreferenced and if they can be deleted.
-             *
-             * Note that the only scenario where there is no reference timestamp is when no ops have ever been processed
-             * for this container and it is in read mode. In this scenario, there is no point in running GC anyway
-             * because references in the container do not change without any ops, i.e., there is nothing to collect.
-             */
-            if (currentReferenceTimestampMs === undefined) {
-                // Log an event so we can evaluate how often we run into this scenario.
-                this.mc.logger.sendErrorEvent({
-                    eventName: "GarbageCollectorInitializedWithoutTimestamp",
-                    gcConfigs: JSON.stringify(this.configs),
-                });
-                return;
-            }
-
-            const baseSnapshotData = await this.baseSnapshotDataP;
```

Comment:
I'm curious behind why you chose to remove this slight optimization and always execute `baseSnapshotDataP`

Reply:
For code organization mostly. The check moved into the `updateStateFromSnapshotData` function since that needs this check as well.
Not having currentReferenceTimestamp is a rare scenario (if it ever happens) so we won't gain much by adding this check here too.

Reply:
Reverted this back as I was working on some other changes.